src/main/java/org/gitlab4j/api/GroupApi.java

```diff
@@ -1576,6 +1577,24 @@ public List<Badge> getBadges(Object groupIdOrPath) throws GitLabApiException {
 	return (response.readEntity(new GenericType<List<Badge>>() {}));
     }
 
+    /**
+     * Gets a list of a group’s badges, case insensitively filtered on name.
+     *
+     * <pre><code>GitLab Endpoint: GET /groups/:id/badges</code></pre>
+     *
+     * @param groupIdOrPath the group ID, path of the group, or a Group instance holding the group ID or path
+     * @param name The name to filter on
+     * @return All badges of the GitLab item, case insensitively filtered on name.
+     * @throws GitLabApiException If any problem is encountered
+     */
+    public List<Badge> getBadges(Object groupIdOrPath, String name) throws GitLabApiException {
```

Comment:
GitLab4j is mostly a wrapper around GitLab API. Actually it seems that there is no possibility to find badges by name. The best way to implement this is to add a filter directly on GitLab. You can create an issue here https://gitlab.com/gitlab-org/gitlab/-/issues
Then when implementing, I will be happy to add the support of this filter.

Reply:
I can remove this until such times as GitLab support it and just have my application do the filter.

Reply:
Have updated my branch to correctly use `?name=foo` as per listing [project](https://docs.gitlab.com/ee/api/project_badges.html#list-all-badges-of-a-project) and [group](https://docs.gitlab.com/ee/api/group_badges.html#list-all-badges-of-a-group) badges.